src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -67,17 +67,32 @@ public override void Initialize(AnalysisContext context)
                     return;
                 }
 
-                startContext.RegisterSyntaxNodeAction(c => AnalyzeNode(c, formatProviderType),
+                startContext.RegisterSyntaxNodeAction(
+                    c => AnalyzeNode(c, formatProviderType),
                     GetInvocationExpressionSyntaxKind());
             });
         }
 
         private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol formatProviderType)
         {
+            var optionSet = context.Options.GetDocumentOptionSetAsync(
+                context.Node.SyntaxTree, context.CancellationToken).GetAwaiter().GetResult();
```

Comment:
As we discussed, there isn't a great way to do this right now.  It is believed the perf hit won't be too bad since currently other analyzers utilize options in the same way.